src/main/java/ch/njol/skript/conditions/base/PropertyCondition.java

```diff
@@ -86,49 +92,66 @@ public enum PropertyType {
 	private Expression<? extends T> expr;
 
 	/**
+	 * Registers a new property condition. The property type is set to {@link PropertyType#BE}.
+	 * 
 	 * @param condition the class to register
 	 * @param property the property name, for example <i>fly</i> in <i>players can fly</i>
 	 * @param type must be plural, for example <i>players</i> in <i>players can fly</i>
 	 */
-
 	public static void register(Class<? extends Condition> condition, String property, String type) {
 		register(condition, PropertyType.BE, property, type);
 	}
 
 	/**
+	 * Registers a new property condition.
+	 * 
 	 * @param condition the class to register
 	 * @param propertyType the property type, see {@link PropertyType}
 	 * @param property the property name, for example <i>fly</i> in <i>players can fly</i>
 	 * @param type must be plural, for example <i>players</i> in <i>players can fly</i>
 	 */
-
 	public static void register(Class<? extends Condition> condition, PropertyType propertyType, String property, String type) {
+		Skript.registerCondition(condition, ConditionType.PROPERTY,
+				getPatterns(propertyType, property, type));
+	}
+
+	/**
+	 * Returns the patterns for the given property type, property and type.
+	 * 
+	 * @param propertyType the property type, see {@link PropertyType}
+	 * @param property the property name, for example <i>fly</i> in <i>players can fly</i>
+	 * @param type must be plural, for example <i>players</i> in <i>players can fly</i>
+	 * @return patterns formmated for {@link Skript#registerCondition(Class, String...)}
+	 */
+	public static String[] getPatterns(PropertyType propertyType, String property, String type) {
```

Comment:
i'd prefer this be private. if someone wants to get the patterns they could use the syntaxelementinfo

Reply:
I'd like it to be public so other syntaxes can use the property patterns without being property conditions, and without duplicating code.

Reply:
![image](https://github.com/user-attachments/assets/7f856d9e-2126-49c0-b455-669bdaef7892)